src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -40,167 +29,142 @@ const distanceToFurthestCorner = (x: number, y: number, rect: ClientRect) => {
+  /** Whether the mouse is currently down or not. */
+  private _isMousedown: boolean = false;
+
+  /** Currently active ripples that will be closed on mouseup. */
+  private _activeRipples: HTMLElement[] = [];
+
+  /** Events to be registered on the trigger element. */
+  private _triggerEvents = new Map<string, any>();
+
+  /** Ripple config for all ripples created by events. */
+  rippleConfig: RippleConfig = {};
+
+  /** Whether mouse ripples should be created or not. */
+  rippleDisabled: boolean = false;
+
+  constructor(_elementRef: ElementRef, private _ngZone: NgZone, private _ruler: ViewportRuler) {
+    this._containerElement = _elementRef.nativeElement;
+
+    // Specify events which need to be registered on the trigger.
+    this._triggerEvents.set('mousedown', this.onMousedown.bind(this));
+    this._triggerEvents.set('mouseup', this.onMouseup.bind(this));
+    this._triggerEvents.set('mouseleave', this.onMouseLeave.bind(this));
+
+    // By default use the host element as trigger element.
+    this.setTriggerElement(this._containerElement);
   }
 
-  /**
-   * Installs event handlers on the given trigger element, and removes event handlers from the
-   * previous trigger if needed.
-   *
-   * @param newTrigger New trigger to which to attach the ripple handlers.
-   */
-  setTriggerElement(newTrigger: HTMLElement) {
-    if (this._triggerElement !== newTrigger) {
-      if (this._triggerElement) {
-        this._eventHandlers.forEach((eventHandler, eventName) => {
-          this._triggerElement.removeEventListener(eventName, eventHandler);
-        });
-      }
-      this._triggerElement = newTrigger;
-      if (this._triggerElement) {
-        this._eventHandlers.forEach((eventHandler, eventName) => {
-          this._triggerElement.addEventListener(eventName, eventHandler);
-        });
-      }
+  /** Fades in a ripple at the given coordinates. */
+  fadeInRipple(pageX: number, pageY: number, config: RippleConfig = {}) {
+    let containerRect = this._containerElement.getBoundingClientRect();
+
+    if (config.centered) {
+      pageX = containerRect.left + containerRect.width / 2;
+      pageY = containerRect.top + containerRect.height / 2;
+    } else {
+      // Subtract scroll values from the coordinates because calculations below
+      // are always relative to the viewport rectangle.
+      let scrollPosition = this._ruler.getViewportScrollPosition();
+      pageX -= scrollPosition.left;
+      pageY -= scrollPosition.top;
     }
-  }
 
-  /** Installs event handlers on the host element of the md-ripple directive. */
-  setTriggerElementToHost() {
-    this.setTriggerElement(this._rippleElement);
-  }
+    let radius = config.radius || distanceToFurthestCorner(pageX, pageY, containerRect);
+    let duration = 1 / (config.speedFactor || 1) * (radius / RIPPLE_SPEED_PX_PER_SECOND);
+    let offsetX = pageX - containerRect.left;
+    let offsetY = pageY - containerRect.top;
+
+    let ripple = document.createElement('div');
```

Comment:
As discussed on Slack, we will revisit this in issue #308 and once we have feedback from Jeremy about this.